### PR TITLE
Fix protocol activation

### LINKED ISSUE #1164: Image for Copy Link Teaching Tip is incorrect for WinUI3
### Which version of the app?

WinUI 3 Gallery

### Description

The image (bellow) for WinUI3 is incorrect, we can see the URI begin with `xamlcontrolgallery://**` but the result link in the application is `winui3gallery://**`.

https://github.com/microsoft/WinUI-Gallery/blob/main/WinUIGallery/Assets/CopyLinkTeachingTip.png

### Screenshots

_No response_

### Windows version

Windows 11 22H2 (22621)

### Additional context

_No response_

### LINKED ISSUE #1300: Deep linking to specific pages does not work
### Which version of the app?

WinUI 3 Gallery

### Description

Deep linking doesn't work.

Steps to reproduce:
- Go to any page
- Click the "Copy link" button.
- Open `Run` prompt, copy-and-paste and hit enter
- WinUI opens, but doesn't navigate to the correct page

Related: #1164

### Screenshots

_No response_

### Windows version

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by chingucoding**
## Description

Simplifies protocol activation,, sorry for the new line changes.
## Motivation and Context

Closes #1300, Closes #1164
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: WinUIGallery/App.xaml.cs [hunk 1]
```diff
@@ -1,288 +1,273 @@
...
+        {
+            this.InitializeComponent();
+
+#if WINUI_PRERELEASE
+            this.Suspending += OnSuspending;
+            this.Resuming += App_Resuming;
+            this.RequiresPointerMode = ApplicationRequiresPointerMode.WhenRequested;
+#endif
+
+            if (ApiInformation.IsApiContractPresent("Windows.Foundation.UniversalApiContract", 6))
```
Comment (niels9001) [inline WinUIGallery/App.xaml.cs:83]: Unrelated.. but we can remove this?

File: WinUIGallery/App.xaml.cs [hunk 2]
```diff
@@ -1,288 +1,273 @@
...
+            }
+            var eventargs = Microsoft.Windows.AppLifecycle.AppInstance.GetCurrent().GetActivatedEventArgs();
+            if (eventargs != null && eventargs.Kind is ExtendedActivationKind.Protocol && eventargs.Data is ProtocolActivatedEventArgs)
+            {
+                ProtocolActivatedEventArgs ProtocolArgs = eventargs.Data as ProtocolActivatedEventArgs;
+                var uri = ProtocolArgs.Uri.LocalPath.Replace("/", "");
+
+                targetPageArguments = uri;
+                string targetId = string.Empty;
+                File.WriteAllText("D:\\uri.txt", uri);
```
Comment (niels9001) [inline WinUIGallery/App.xaml.cs:188]: Debug code :)? This crashes the app on my machine as it's not able to find this file

File: WinUIGallery/App.xaml.cs [hunk 3]
```diff
@@ -1,288 +1,273 @@
...
+                {
+                    targetPageType = typeof(SectionPage);
+                }
+                else if (ControlInfoDataSource.Instance.Groups.Any(g => g.Items.Any(i => i.UniqueId == uri)))
+                {
+                    targetPageType = typeof(ItemPage);
+                }
+            }
+
+
```
Comment (niels9001) [inline WinUIGallery/App.xaml.cs:208]: Nit: empty line

File: WinUIGallery/App.xaml.cs [hunk 4]
```diff
@@ -1,288 +1,273 @@
...
+                {
+                    targetPageType = typeof(AllControlsPage);
+                }
+                else if (uri == "NewControls")
+                {
+                    targetPageType = typeof(NewControlsPage);
+                }
+                else if (ControlInfoDataSource.Instance.Groups.Any(g => g.UniqueId == uri))
+                {
+                    targetPageType = typeof(SectionPage);
```
Comment (niels9001) [inline WinUIGallery/App.xaml.cs:200]: Nit: `NavigationView` selection doesn't seem to work for e.g. the Colors/Icons pages. Not a super big deal though, so nothing blocking.

image

**Review by niels9001 — CHANGES_REQUESTED** [review]
1 crashing bug and a few nits.

Would be good to replace the `CopyLinkTeachingTip.png` with a new image as well:

CopyLinkTeachingTip

File: WinUIGallery/App.xaml.cs (hunk 4 above)
Comment (chingucoding) [reply WinUIGallery/App.xaml.cs:194]: Good catch, we can also fix this in this PR if you prefer. That way this topic should be done for some time then.

Comment (niels9001) [reply WinUIGallery/App.xaml.cs:194]: Sure!

**Comment by niels9001** [comment]
@chingucoding Oops, saw that exporting wasn't successful with the previous image. I've removed the blinking cursor in this image:

**Comment by chingucoding** [comment]
I think this should be everything now, protocol activation should work for all pages and the image is up-to-date.

**Review by pratikone — CHANGES_REQUESTED** [review]
it seems that your changes include file encoding changes. could you revert those. github and VS keep doing funky things around it and we are trying to keep it in the old state until we find a reliable solution. so that files are not getting modified entirely from time to time.

**Comment by chingucoding** [comment]
Should be good now @pratikone, really hope we can find a solution for this at some point.

**Comment by Jay-o-Way** [comment]
What's keeping this?

**Review by pratikone — APPROVED** [review]
thanks again for making winui gallery better. Really helps the community 💖

**Comment by Jay-o-Way** [comment]
@chingucoding those squash descriptions include what seems to be personal e-mail addresses. Is that alright to be public?

**Comment by chingucoding** [comment]
> @chingucoding those squash descriptions include what seems to be personal e-mail addresses. Is that alright to be public?

Thanks for mentioning, this is a valid point. However the squash commits do is, well squash the commits. The summary is just a list of the commit details so all of that information can also be looked up with a `git log` on this repository. If I or others should use their email or have GitHub mask this is another matter though 😅

**Review by niels9001 — APPROVED** [review]
LGTM!

**Pull request merged**